apollo-router/src/query_planner/caching_query_planner.rs

```diff
@@ -24,25 +22,24 @@ type PlanResult = Result<QueryPlannerContent, QueryPlannerError>;
 /// A query planner wrapper that caches results.
 ///
 /// The query planner performs LRU caching.
-#[derive(Debug)]
-pub(crate) struct CachingQueryPlanner<T: QueryPlanner> {
-    cm: Arc<CachingMap<QueryKey, QueryPlannerContent>>,
-    phantom: PhantomData<T>,
+#[derive(Clone)]
+pub(crate) struct CachingQueryPlanner<T: QueryPlanner + Clone> {
+    cm: Arc<DeduplicatingCache<QueryKey, Result<QueryPlannerContent, QueryPlannerError>>>,
```

Comment:
NIT: cm was short for `CachingMap`, should be re-named.

Reply:
fixed in 50016965